tests/test_azure.py

```diff
@@ -514,6 +514,45 @@ def test_upload_to_container_blob_from_storage(self,
         mock_callback.assert_called_once()
         self.assertEqual(res, mock_blob)
 
+    @patch("cloudimg.ms_azure.AzureService.are_tags_present")
+    @patch("cloudimg.ms_azure.AzureService.get_container_by_name")
+    def test_copy_to_conainer_from_url(
+        self,
+        mock_get_container_by_name,
+        mock_are_tags_present,
+    ):
+        mock_blob_client = MagicMock()
+        mock_blob_client.exists.return_value = False
+        mock_container_client = MagicMock()
+        mock_container_client.get_blob_client.return_value = mock_blob_client
+        mock_get_container_by_name.return_value = mock_container_client
+        mock_are_tags_present.return_value = False
+
+        mock_blob_client.get_blob_properties.side_effect = [
+            {"copy": {"status": "Pending", "progress": "0/2"}},
+            {"copy": {"status": "Pending", "progress": "1/2"}},
+            {"copy": {"status": "success", "progress": "2/2"}},
+        ]
```

Comment:
I suggest you to add a test covering the unsuccessful case as well.  Relates to what I said about, I think, the unsuccessful case currently not raising an exception.